src/Core/src/Handlers/View/ViewHandler.cs

```diff
@@ -131,6 +132,24 @@ public static void MapAutomationId(IViewHandler handler, IView view)
 			((NativeView?)handler.NativeView)?.UpdateAutomationId(view);
 		}
 
+		public static void MapClipShape(IViewHandler handler, IView view)
+		{
+			var clipShape = view.ClipShape;
+
+			if (clipShape != null)
+				handler.HasContainer = true;
+			else
+			{
+				if (handler is ViewHandler viewHandler)
+					handler.HasContainer = viewHandler.NeedsContainer;
+			}
+#if WINDOWS			
+			((NativeView?)handler.ContainerView)?.UpdateClipShape(view);
```

Comment:
We can probably add a `using WrapperView = Microsoft.UI.Xaml.Controls.Border;` at the top. Real pity we can't inherit from that being sealed and all.